File: excel-api-java/src/test/java/pl/alyx/api/excel/logging/JsonLogFormatTest.java
package pl.alyx.api.excel.logging;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class JsonLogFormatTest {

    @Test
    public void testLogEntryFieldNames() {
        LogEntry logEntry = new LogEntry();
        logEntry.level = "info";
        logEntry.date = "2025-03-15";
        logEntry.time = "14:24:58.123";
        logEntry.message = "Request completed";

        assertNotNull(logEntry.level);
        assertNotNull(logEntry.date);
        assertNotNull(logEntry.time);
        assertNotNull(logEntry.message);

        // Verify date format YYYY-MM-DD
        assertTrue(logEntry.date.matches("^\\d{4}-\\d{2}-\\d{2}$"));

        // Verify time format HH:mm:ss.SSS
        assertTrue(logEntry.time.matches("^\\d{2}:\\d{2}:\\d{2}\\.\\d{3}$"));
    }

    private static class LogEntry {
        public String level;
        public String date;
        public String time;
        public String message;
    }
}

File: excel-api-csharp/src/ExcelApi.Test/JsonLogFormatTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Text.RegularExpressions;

namespace ExcelApi.Tests;

[TestClass]
public class JsonLogFormatTest
{
    [TestMethod]
    public void TestLogEntryFieldNames()
    {
        var logData = new
        {
            level = "info",
            date = "2025-03-15",
            time = "14:24:58.123",
            message = "Request completed",
            request = new
            {
                method = "GET",
                url = "/api/v1/workbooks"
            },
            response = new
            {
                statusCode = 200
            },
            remote = "127.0.0.1"
        };

        Assert.IsNotNull(logData.level);
        Assert.IsNotNull(logData.date);
        Assert.IsNotNull(logData.time);
        Assert.IsNotNull(logData.message);
        Assert.IsNotNull(logData.request);
        Assert.IsNotNull(logData.response);
        Assert.IsNotNull(logData.remote);

        // Verify date format YYYY-MM-DD
        StringAssert.Matches(logData.date, new Regex(@"^\d{4}-\d{2}-\d{2}$"));

        // Verify time format HH:mm:ss.fff
        StringAssert.Matches(logData.time, new Regex(@"^\d{2}:\d{2}:\d{2}\.\d{3}$"));
    }
}
